frontend/pages/index.tsx

```diff
@@ -30,17 +36,36 @@ export const getStaticProps: GetStaticProps = async () => {
         }
       },
       cheftext,
-      chefname
+      chefname,
+    }
+  `;
+  const galleryWidgetQuery = `
+  *[_type == "HomePage"]{
+      galleryPhotos[]{
+        asset -> {
+        url
+        }
+      },
+      menuName,
+      menuLink,
+      menuDescription[]{
+        children[]{
+          text
+        }
+      }
     }
   `;
 
-  const homePageContent = await sanityClient.fetch(query);
+  const homePageContent = await Promise.all([
+    sanityClient.fetch(heroQuery),
+    sanityClient.fetch(galleryWidgetQuery),
```

Comment:
I believe we can combine the two client fetch into one and then deconstruct it. I believe it should be ok to fetch all the content at least for homepage at once instead of making two calls we can make this one call, and this would make our cost more effectively
```
const query = `
    *[_type == "HomePage"][0]{
      Homepagetitle,
      "backgroundImgUrl": backgroundimg.asset->url,
      "dishImgUrl": dishimg.asset->url,
      cheftext,
      chefname,
      "galleryPhotos": galleryPhotos[]{
        "url": asset->url
      },
      menuName,
      menuLink,
      "menuDescription": menuDescription[0].children[0].text
    }
  `;

  try {
    const data = await client.fetch(query);

    return {
      props: {
        heroData: {
          Homepagetitle: data.Homepagetitle,
          backgroundImgUrl: data.backgroundImgUrl,
          dishImgUrl: data.dishImgUrl,
          cheftext: data.cheftext,
          chefname: data.chefname,
        },
        galleryWidgetData: {
          galleryPhotos: data.galleryPhotos,
          menuName: data.menuName,
          menuLink: data.menuLink,
          menuDescription: data.menuDescription,
        },
      },
    };
  } catch (error) {
    console.error('Error fetching data:', error);
    return {
      props: {
        heroData: null,
        galleryWidgetData: null,
      },
    };
  }
};

Reply:
Something like this can work but you need to have a test, this is my brief guessing

Reply:
updated